contracts/IColonyNetwork.sol

```diff
@@ -183,12 +183,34 @@ contract IColonyNetwork {
   function setupRegistrar(address _ens, bytes32 _rootNode) public;
 
   /// @notice Register a "user.joincolony.eth" label.
-  /// @param subnode The keccak256 hash of the label to register.
-  function registerUserLabel(bytes32 subnode) public;
+  /// @param username The label to register
+  /// @param orbitdb  The path of the orbitDB database associated with the user profile
+  function registerUserLabel(string username, string orbitdb) public;
 
   /// @notice Register a "colony.joincolony.eth" label. Can only be called by a Colony.
-  /// @param subnode The keccak256 hash of the label to register.
-  function registerColonyLabel(bytes32 subnode) public;
+  /// @param colonyName The label to register.
+  function registerColonyLabel(string colonyName) public;
+
+  /// @notice Retrieve the orbitdb address corresponding to a registered account
+  /// @param node The Namehash of the account being queried.
+  /// @return orbitDB A string containing the address of the orbit database
+  function getProfileDBAddress(bytes32 node) public view returns (string orbitDB);
+
+  /// @notice Reverse lookup a username from an address.
+  /// @param addr The address we wish to find the corresponding ENS domain for (if any)
+  /// @return domain A string containing the colony-based ENS name corresponding to addr
+  function lookupUsername(address addr) public view returns(string domain);
+
+  /// @notice Query if a contract implements an interface
+  /// @param interfaceID The interface identifier, as specified in ERC-165
+  /// @dev Interface identification is specified in ERC-165.
+  /// @return `true` if the contract implements `interfaceID`
+  function supportsInterface(bytes4 interfaceID) public view returns (bool);
```

Comment:
We can make this an `external` function as specified in the ERC but also because it'll be cheaper. (Note to self we should review our function visibility modifiers across everything)